Issue #357 - CLI --testPathPattern option

McNouvion opened the pull request:
Let me know if you need any changes or have any comments.
I did not think this deserved a specific test since I feel it is an edge case but let me know if you feel otherwise.

Thanks!

DmitrySoshnikov commented:
I see, I guess you're right, in case we want a real regexp, it should no espace `(` and `)`, so now I think we don't need `escapeStrForRegex` :smile: Also, we should keep in mind, that `new RegExp` expects `\s` and similar to be `\\s`. We can leave it up to a user, but seems would be better to handle it, and append those slashes.

```javascript
(new RegExp('\sa')).test(' a'); // false
(new RegExp('\\sa')).test(' a'); // true
```

McNouvion commented:
I don't have the right project setup at home to test this properly so I will do some more testing on Monday.

McNouvion commented:
I have checked and you are correct `argv.testPathPattern` is already escaped for slashes. So if we wanted to add some safety checks it would have to happen earlier in the process ( In optimist args check?). Anyways, I have updated the code and also added the params help info in Bin/jest.js as you originally requested. Let me know if you need me to make any updates, thanks!

DmitrySoshnikov commented:
OK, looks great, thanks!
